core-rust/state-manager/src/state_manager.rs

```diff
@@ -341,6 +346,23 @@ where
         .execute_on(&self.store))
     }
 
+    /// Adds the given transaction to the mempool (applying all the checks and caching, see
+    /// `check_for_rejection_and_add_to_mempool()`), and then triggers an unscheduled mempool sync
+    /// (meant to relay the new transaction submitted via Core API).
+    pub fn add_to_mempool_and_trigger_relay(
+        &mut self,
+        transaction: NotarizedTransaction,
+    ) -> Result<(), MempoolAddError> {
+        self.check_for_rejection_and_add_to_mempool(
+            MempoolAddSource::CoreApi,
+            transaction.clone(),
+        )?;
+        if let Err(error) = self.mempool_relay_dispatcher.trigger_relay(transaction) {
```

Comment:
Is it worth adding a comment that we only send MempoolAdd messages when we add the transaction to our mempools? (ie it's valid and not a duplicate) - to prevent infinite flood fill.